core/mixins.py

```diff
@@ -0,0 +1,32 @@
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from datetime import datetime
+from app import db
+
+# Creating the customer model
+class BaseModelMixen(object):
+    id = db.Column(db.Integer, primary_key=True)
+    date_created = db.Column(db.DateTime, default=datetime.utcnow)
+    # Nullable default is true, Unique default is False
+
+    def save(self):
+        # Used to save data
+        db.session.add(self)
+        return db.session.commit()
+
+    def delete(self):
+        # Used to delete data
+        db.session.delete(self)
+        return db.session.commit()
+
+    def to_dict(self):
```

Comment:
Question: How would this method work if you end up using `BaseModelMixin` to define another model someday?

Read what I said above about how model dependencies should flow.

I'm definitely going to have you making more models and views and they'll all inherit from `BaseModelMixin`.

Reply:
Hint: you can define a base method here that you can always override or extend, maybe with something like `super().save()`?